packages/angular_devkit/build_angular/src/utils/index-file/add-js-action-contract.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { readFile } from 'node:fs/promises';
+import { htmlRewritingStream } from './html-rewriting-stream';
+
+let jsActionContractScript: string;
+
+export async function addJsActionContract(html: string): Promise<string> {
+  const { rewriter, transformedContent } = await htmlRewritingStream(html);
+
+  jsActionContractScript ??=
+    '<script type="text/javascript" name="ng-jsa-contract">' +
```

Comment:
@AndrewKushnir, I used a name here do the possibility that an ID would conflict if there are 2 NG apps running on the same, and unfortunately at this point we don't have access to the API ID.

Reply:
@clydin mentioned that JSA in 3P has been renamed to event-delegation. Should we change this to something else instead of JSA?

Reply:
@alan-agius4 this script should only be present on a page once (irrespective of the number of apps on a page), so I think using an "id" attribute should be ok.

> Should we change this to something else instead of JSA?

We can call it "ng-event-contract" instead.

Reply:
> this script should only be present on a page once (irrespective of the number of apps on a page)

Just to confirm: would that be the case with the change in this PR?

Reply:
Also cc @iteriani for visibility and context.

Reply:
> Just to confirm: would that be the case with the change in this PR?

Yes